documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc

```diff
@@ -1022,13 +1025,33 @@ Next, functions for working with numeric values:
 | `exp()` | Exponential function | `exp(x)` | Universal in SQL dialects
 | `power()` | Exponentiation | `power(x,y)` | Universal in SQL dialects
 | `ln()` | Natural logarithm | `ln(x)` | Very common in SQL dialects
+| `log()` | Base-N logarithm | `log(x,x)` | Very common in SQL dialects
+| `log2()` | Base-2 logarithm | `log2(x)` | Very common in SQL dialects
 | `log10()` | Base-10 logarithm | `log10(x)` | Very common in SQL dialects
-| `sin()`, `cos()`, `tan()`, `asin()`, `acos()`, `atan()`, `atan2()` | Basic trigonometric functions | `sin(theta)`, `cos(theta)`, `atan2(opposite, adjacent)` | Very common in SQL dialects
+| `sin()`, `cos()`, `tan()`, `cot()`, `asin()`, `acos()`, `atan()`, `atan2()`, `acot()` | Basic trigonometric functions | `sin(theta)`, `cos(theta)`, `atan2(opposite, adjacent)` | Very common in SQL dialects
+| `sinh()`, `cosh()`, `tanh()`, `coth()`, `asinh()`, `acosh()`, `atanh()`, `acoth()` | Basic trigonometric functions | `sinh(theta)`, `cosh(theta)` | Very common in SQL dialects
 | `round()` | Numeric rounding | As usual: `round(number, places)` | Very common in SQL dialects
 | `floor()` | Floor function | `floor(x)` | Universal in SQL dialects
 | `ceiling()` | Ceiling function | `ceiling(x)` | Very common in SQL dialects
 | `least()` | Return the smallest of the given arguments | `least(x, y, z)` | Very common in SQL dialects
 | `greatest()` | Return the largest of the given arguments | `greatest(x, y, z)` | Very common in SQL dialects
+| `degrees()` | Converts radians to degrees | `degrees(x)` | Very common in SQL dialects
+| `radians()` | Converts degrees to radians | `radians(x)` | Very common in SQL dialects
+| `pi()` | The pi constant | `pi()` | Very common in SQL dialects
+| `rand()` | Return random double between 0 and 1 | `rand()` | Very common in SQL dialects
```

Comment:
Rollback please.

Reply:
1. `pi()` is not supported on all built-in dialects, but it can be emulated using `asin(1.0)`.
2. `degrees()` and `radians()` aren't supported on all built-in dialects, but I guess they can be emulated using `/180*asin(1.0)`.
3. `rand()` is not supported on all built-in dialects, and making it portable might be tricky, I'm not sure.

None of these functions belong on the list unless someone does the work to make them work portably.